lab_3_generate_by_ngrams/main.py

```diff
@@ -93,6 +124,21 @@ def encode(self, text: str) -> Optional[tuple[int, ...]]:
         In case of corrupt input arguments, None is returned.
         In case any of methods used return None, None is returned.
         """
+        if not isinstance(text, str) or len(text) == 0:
+            return None
+        encoded_dictionary = {}
+        encoded_corpus = []
+        tokens = self._tokenize(text)
+        if tokens is None:
+            return None
+        for token in tokens:
+            encoded_dictionary[token] = self._put(token)
```

Comment:
For what reason do you need a dictionary with tokens, where all the values are `None`?

Reply:
Fixed. I tried to do it as in the example of _put but now I understand that it's used nowhere and it isn't necessary